client/my-sites/post-type-list/list.jsx

```diff
@@ -0,0 +1,95 @@
+/**
+ * External dependencies
+ */
+import React, { Component, PropTypes } from 'react';
+import { connect } from 'react-redux';
+import omit from 'lodash/omit';
+
+/**
+ * Internal dependencies
+ */
+import InfiniteList from 'components/infinite-list';
+import { getSelectedSiteId } from 'state/ui/selectors';
+import {
+	isRequestingSitePostsForQuery,
+	isSitePostsLastPageForQuery,
+	getSitePostsForQueryIgnoringPage
+} from 'state/posts/selectors';
+import PostTypeListPost from './post';
+import PostTypeListPostPlaceholder from './post-placeholder';
+import PostTypeListEmptyContent from './empty-content';
+
+/**
+ * Constants
+ */
+const GUESSED_POST_HEIGHT = 86;
+
+class PostTypeListList extends Component {
+	getListKey() {
+		const { siteId, query } = this.props;
+		return [ siteId, JSON.stringify( omit( query, 'page' ) ) ].join();
+	}
+
+	getPostItemRef( post ) {
+		return post.global_ID;
+	}
+
+	renderPostItem( post ) {
+		return (
+			<PostTypeListPost
+				key={ post.global_ID }
+				globalId={ post.global_ID } />
+		);
+	}
+
+	renderLoadingPlaceholder() {
+		return (
+			<PostTypeListPostPlaceholder key="placeholder" />
+		);
+	}
+
+	render() {
+		const { query, posts, lastPage, requesting, requestNextPage } = this.props;
+
+		return (
+			<div className="post-type-list__list">
+				{ posts && ! posts.length && ! requesting && (
+					<PostTypeListEmptyContent
+						type={ query.type }
+						status={ query.status } />
+				) }
+				<InfiniteList
+					key={ this.getListKey() }
+					items={ posts || [] }
```

Comment:
>Any reason to not set a default prop of `[]` for `posts`?

Hmm, while I think we could, an empty array `posts` has a different meaning than `null` posts, which is that we only want to consider posts as an empty array if we've received an empty result from the REST API. This is relevant a few lines above in determining whether to render `<PostTypeListEmptyContent />` only if we know that there are no posts (vs. not yet received from the API). That said, we'd at least be able to fall back on `requesting` always being true when posts are unknown.

I'm inclined to keep it as it is, since otherwise we might run into issues if we need to rely on there being this distinction.